### PR TITLE
Shape loader fixes

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Fices #95

### CODE REVIEW

File: src/models/shape_loader.cpp [hunk 1]
```diff
@@ -893,9 +911,9 @@ geo::ShapePtr loadShape(const std::string& model_path, tue::config::Reader cfg,
         error << "[ED::MODELS::LOADSHAPE] Error while loading shape with data:" << std::endl << cfg.data() << std::endl;
     }
 
+    // Extra pose is only allowed in ED yaml.
     readPose(cfg, pose);
-
-    if (shape)
+    if (shape && pose != geo::Pose3D::identity())
```
Comment (PetervDooren) [inline src/models/shape_loader.cpp:916]: This solves issue https://github.com/tue-robotics/ed/issues/95 . However the changes made here are a workaround. I am not saying that you should change it but know that the issue still persists in the case of ED yamls. And when people encounter the problem they might not immediatly spot what is going wrong.

File: src/models/shape_loader.cpp [hunk 2]
```diff
@@ -732,20 +738,29 @@ geo::ShapePtr loadShape(const std::string& model_path, tue::config::Reader cfg,
...
     {
         int num_points = 12;
         cfg.value("num_points", num_points, tue::config::OPTIONAL);
 
         double radius, height;
-        if (cfg.value("radius", radius) && (cfg.value("height", height) || cfg.value("length", height))) //length is used in SDF
+        if (cfg.value("radius", radius) && cfg.value("height", height)) //length is used in SDF
         {
+            cfg.endGroup();
+            readPose(cfg, pose);
```
[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply src/models/shape_loader.cpp:752]: This pose is used in the end. If in the end no pose is read(readPose is trying, but doesn't find something to read), this pose is used. To my knowledge max one of the two pose are present. Because reading one in the end would override the first one. To be 100% robust, we could create an extra pose. And check on the product of the two poses.

File: src/models/shape_loader.cpp (hunk 1 above)
Comment (MatthijsBurgh) [reply src/models/shape_loader.cpp:916]: True, but to completely get rid of this for the ED yamls too, we would need to break the current 'spec', which I don't want, because we are trying to move away from it.

File: src/models/shape_loader.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (PetervDooren) [reply src/models/shape_loader.cpp:752]: But in line 915 we read the pose again. So we perform readpose twice without using the result. This operation is performed at the same level of cfg. So what exactly went wrong in the previous version of this piece of code (apart from the missing cfg.endGroup())

[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply src/models/shape_loader.cpp]: This was indeed not an improvement, so I reverted this part of the changes.

File: src/models/model_loader.cpp [hunk 3]
```diff
@@ -1,5 +1,7 @@
 #include "ed/models/model_loader.h"
 
+#include <boost/make_shared.hpp>
```
Comment (PetervDooren) [inline src/models/model_loader.cpp:3]: Why is this added? Was something broken? Please elaborate

File: .travis.yml [hunk 4]
```diff
@@ -1,5 +1,5 @@
-sudo: true
-
+os: linux
+dist: xenial
```
Comment (PetervDooren) [inline .travis.yml:2]: Please elaborate as I am not too familiar with travis. Why was this change made?

Comment (MatthijsBurgh) [reply .travis.yml:2]: Travis was producing warnings, If not defined, travis assumed these values, but better to make it explicit.

File: src/models/model_loader.cpp (hunk 3 above)
Comment (MatthijsBurgh) [reply src/models/model_loader.cpp:3]: I experienced this being a problem on my computer. I have a more changes on my computer which remove some boost dependencies where it wasn't needed. That showed the missing include here. Use here: https://github.com/tue-robotics/ed/blob/ce221ae162e32930f435e68412b59b572a84ae85/src/models/model_loader.cpp#L545

File: .travis.yml (hunk 4 above)
Comment (PetervDooren) [reply .travis.yml:2]: thanks

**Comment by PetervDooren** [comment]
I believe these changes do not require a test on the real robot. So if you agree with that Matthijs you may merge this.
